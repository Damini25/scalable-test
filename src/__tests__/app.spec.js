import React from 'react';
import { render, cleanup, act } from '@testing-library/react';
import App from '../app';
import cones from '../../cones.json';

afterEach(cleanup)
test('render <App>', () => {
    const wrapper = render(<App />)
    expect(wrapper).toMatchSnapshot()
})
test('test for fetchConesData api call', async () => {
    const wrapper = render(<App />)
    jest.spyOn(global, "fetch").mockImplementation(() =>
        Promise.resolve({
            json: () => Promise.resolve(cones)
        })
    );
    await act(async () => {
        render(<App />);
      });
})
