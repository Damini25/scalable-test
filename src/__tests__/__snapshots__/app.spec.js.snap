// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <App> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <p>
            <a
              href="/table"
            >
              Table
            </a>
          </p>
          <p>
            <a
              href="/chart"
            >
              Chart
            </a>
          </p>
        </div>
        <div>
          <div>
            <label>
              Risk level:
            </label>
            <select
              data-testid="risk-options-id"
              name="riskLevel"
            >
              <option
                value="3"
              >
                3
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
              <option
                value="6"
              >
                6
              </option>
              <option
                value="7"
              >
                7
              </option>
              <option
                value="8"
              >
                8
              </option>
              <option
                value="9"
              >
                9
              </option>
              <option
                value="10"
              >
                10
              </option>
              <option
                value="11"
              >
                11
              </option>
              <option
                value="12"
              >
                12
              </option>
              <option
                value="13"
              >
                13
              </option>
              <option
                value="14"
              >
                14
              </option>
              <option
                value="15"
              >
                15
              </option>
              <option
                value="16"
              >
                16
              </option>
              <option
                value="17"
              >
                17
              </option>
              <option
                value="18"
              >
                18
              </option>
              <option
                value="19"
              >
                19
              </option>
              <option
                value="20"
              >
                20
              </option>
              <option
                value="21"
              >
                21
              </option>
              <option
                value="22"
              >
                22
              </option>
              <option
                value="23"
              >
                23
              </option>
              <option
                value="24"
              >
                24
              </option>
              <option
                value="25"
              >
                25
              </option>
            </select>
          </div>
          <div>
            <label>
              No. of years:
            </label>
            <input
              data-testid="years-id"
              max="1000"
              min="1"
              name="years"
              type="number"
              value="1"
            />
          </div>
          <div>
            <label>
              Initial Investment Sum:
            </label>
            <input
              max="10000"
              min="1"
              name="investmentSum"
              type="number"
              value="1000"
            />
          </div>
          <div>
            <label>
              Initial Monthly Sum:
            </label>
            <input
              max="10000"
              min="1"
              name="monthlySum"
              type="number"
              value="200"
            />
          </div>
        </div>
        <table>
          <thead>
            <tr>
              <th>
                month
              </th>
              <th>
                good
              </th>
              <th>
                median
              </th>
              <th>
                bad
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                0
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                1
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                2
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                3
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                4
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                5
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                6
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                7
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                8
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                9
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                10
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                11
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
            <tr>
              <td>
                12
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
              <td>
                NaN
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <p>
          <a
            href="/table"
          >
            Table
          </a>
        </p>
        <p>
          <a
            href="/chart"
          >
            Chart
          </a>
        </p>
      </div>
      <div>
        <div>
          <label>
            Risk level:
          </label>
          <select
            data-testid="risk-options-id"
            name="riskLevel"
          >
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="11"
            >
              11
            </option>
            <option
              value="12"
            >
              12
            </option>
            <option
              value="13"
            >
              13
            </option>
            <option
              value="14"
            >
              14
            </option>
            <option
              value="15"
            >
              15
            </option>
            <option
              value="16"
            >
              16
            </option>
            <option
              value="17"
            >
              17
            </option>
            <option
              value="18"
            >
              18
            </option>
            <option
              value="19"
            >
              19
            </option>
            <option
              value="20"
            >
              20
            </option>
            <option
              value="21"
            >
              21
            </option>
            <option
              value="22"
            >
              22
            </option>
            <option
              value="23"
            >
              23
            </option>
            <option
              value="24"
            >
              24
            </option>
            <option
              value="25"
            >
              25
            </option>
          </select>
        </div>
        <div>
          <label>
            No. of years:
          </label>
          <input
            data-testid="years-id"
            max="1000"
            min="1"
            name="years"
            type="number"
            value="1"
          />
        </div>
        <div>
          <label>
            Initial Investment Sum:
          </label>
          <input
            max="10000"
            min="1"
            name="investmentSum"
            type="number"
            value="1000"
          />
        </div>
        <div>
          <label>
            Initial Monthly Sum:
          </label>
          <input
            max="10000"
            min="1"
            name="monthlySum"
            type="number"
            value="200"
          />
        </div>
      </div>
      <table>
        <thead>
          <tr>
            <th>
              month
            </th>
            <th>
              good
            </th>
            <th>
              median
            </th>
            <th>
              bad
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              0
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              1
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              3
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              4
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              5
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              6
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              7
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              8
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              10
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              11
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
          <tr>
            <td>
              12
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
            <td>
              NaN
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
